From 20b4e35899b3698b8c621e7f4b90699259880fc1 Mon Sep 17 00:00:00 2001
From: sreenivasa murthy kolam <sreenivasamurthy.kolam@amd.com>
Date: Wed, 31 May 2023 10:11:26 +0000
Subject: [PATCH] Build with out libhsa-amd-aqlprofile64.so in spack for now as
 Aqlprofile is a closed source and needs a new recipe for downloading the
 library

---
 cmake_modules/env.cmake                 |  2 +-
 src/api/CMakeLists.txt                  | 10 ++++++++--
 src/tools/rocprofv2/CMakeLists.txt      |  9 +++++++--
 tests/unittests/core/CMakeLists.txt     | 13 +++++++++----
 tests/unittests/profiler/CMakeLists.txt | 16 +++++++++++-----
 5 files changed, 36 insertions(+), 14 deletions(-)

diff --git a/cmake_modules/env.cmake b/cmake_modules/env.cmake
index 6647578..70b411e 100644
--- a/cmake_modules/env.cmake
+++ b/cmake_modules/env.cmake
@@ -75,5 +75,5 @@ endif ()
 
 find_library ( FIND_AQL_PROFILE_LIB "libhsa-amd-aqlprofile64.so" HINTS ${CMAKE_INSTALL_PREFIX} PATHS ${ROCM_ROOT_DIR})
 if (  NOT FIND_AQL_PROFILE_LIB )
-  message ( FATAL_ERROR "AQL_PROFILE not installed. Please install AQL_PROFILE" )
+  message ( "AQL_PROFILE not installed. Please install AQL_PROFILE" )
 endif()
diff --git a/src/api/CMakeLists.txt b/src/api/CMakeLists.txt
index 822ee41..95fc4e0 100644
--- a/src/api/CMakeLists.txt
+++ b/src/api/CMakeLists.txt
@@ -40,7 +40,7 @@ get_filename_component(HSA_RUNTIME_INC_PATH ${HSA_H} DIRECTORY)
 find_library(AQLPROFILE_LIB "libhsa-amd-aqlprofile64.so" HINTS ${CMAKE_PREFIX_PATH} PATHS ${ROCM_PATH} PATH_SUFFIXES lib)
 
 if(NOT AQLPROFILE_LIB)
-  message(FATAL_ERROR "AQL_PROFILE not installed. Please install hsa-amd-aqlprofile!")
+  message("AQL_PROFILE not installed. Please install hsa-amd-aqlprofile!")
 endif()
 
 # ############################################################################################################################################
@@ -241,7 +241,13 @@ if(ASAN)
   target_link_libraries(${ROCPROFILER_TARGET} PRIVATE ${AQLPROFILE_LIB} hsa-runtime64::hsa-runtime64 Threads::Threads atomic asan dl c stdc++ stdc++fs amd_comgr ${PCIACCESS_LIBRARIES})
 else()
   target_link_options(${ROCPROFILER_TARGET} PRIVATE -Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/exportmap -Wl,--no-undefined)
-  target_link_libraries(${ROCPROFILER_TARGET} PRIVATE ${AQLPROFILE_LIB} hsa-runtime64::hsa-runtime64 Threads::Threads atomic dl c stdc++ stdc++fs amd_comgr ${PCIACCESS_LIBRARIES})
+  if(NOT AQLPROFILE_LIB)
+    message("AQL_PROFILE not installed. Please install hsa-amd-aqlprofile!")
+    target_link_libraries(${ROCPROFILER_TARGET} PRIVATE  hsa-runtime64::hsa-runtime64 Threads::Threads atomic dl c stdc++ stdc++fs amd_comgr ${PCIACCESS_LIBRARIES})
+  else()
+    target_link_libraries(${ROCPROFILER_TARGET} PRIVATE ${AQLPROFILE_LIB} hsa-runtime64::hsa-runtime64 Threads::Threads atomic dl c stdc++ stdc++fs amd_comgr ${PCIACCESS_LIBRARIES})
+  endif()
+
 endif()
 
 ## Install libraries: Non versioned lib file in dev package
diff --git a/src/tools/rocprofv2/CMakeLists.txt b/src/tools/rocprofv2/CMakeLists.txt
index b2dc968..6cb0073 100644
--- a/src/tools/rocprofv2/CMakeLists.txt
+++ b/src/tools/rocprofv2/CMakeLists.txt
@@ -6,7 +6,7 @@ get_property(HSA_RUNTIME_INCLUDE_DIRECTORIES TARGET hsa-runtime64::hsa-runtime64
 
 find_library(AQLPROFILE_LIB "libhsa-amd-aqlprofile64.so" HINTS ${CMAKE_PREFIX_PATH} PATHS ${ROCM_PATH} PATH_SUFFIXES lib)
 if(NOT AQLPROFILE_LIB)
-  message(FATAL_ERROR "AQL_PROFILE not installed. Please install hsa-amd-aqlprofile!")
+  message("AQL_PROFILE not installed. Please install hsa-amd-aqlprofile!")
 endif()
 
 file(GLOB ROCPROFV2_SRC_FILES ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp)
@@ -23,7 +23,12 @@ target_include_directories(rocprofv2
   PRIVATE
     ${PROJECT_SOURCE_DIR}
     ${PROJECT_SOURCE_DIR}/inc)
+
+if(NOT AQLPROFILE_LIB)
+target_link_libraries(rocprofv2 PRIVATE hsa-runtime64::hsa-runtime64 stdc++fs Threads::Threads atomic -ldl)
+else()
 target_link_libraries(rocprofv2 PRIVATE ${AQLPROFILE_LIB} hsa-runtime64::hsa-runtime64 stdc++fs Threads::Threads atomic -ldl)
+endif()
 # install(TARGETS rocprofv2 RUNTIME
 #   DESTINATION ${CMAKE_INSTALL_BINDIR}
-#   COMPONENT runtime)
\ No newline at end of file
+#   COMPONENT runtime)
diff --git a/tests/unittests/core/CMakeLists.txt b/tests/unittests/core/CMakeLists.txt
index 5efc625..5505d18 100644
--- a/tests/unittests/core/CMakeLists.txt
+++ b/tests/unittests/core/CMakeLists.txt
@@ -96,11 +96,16 @@ target_include_directories(runCoreUnitTests PRIVATE ${PROJECT_SOURCE_DIR}
 target_compile_definitions(runCoreUnitTests
   PUBLIC AMD_INTERNAL_BUILD
   PRIVATE PROF_API_IMPL HIP_PROF_HIP_API_STRING=1 __HIP_PLATFORM_AMD__=1)
-
+if(NOT AQLPROFILE_LIB )
 # Link test executable against gtest & gtest_main
-target_link_libraries(runCoreUnitTests PRIVATE ${ROCPROFILER_TARGET} ${AQLPROFILE_LIB}
-  hsa-runtime64::hsa-runtime64 c stdc++
-  GTest::gtest GTest::gtest_main stdc++fs dl ${PCIACCESS_LIBRARIES})
+  target_link_libraries(runCoreUnitTests PRIVATE ${ROCPROFILER_TARGET}
+    hsa-runtime64::hsa-runtime64 c stdc++
+    GTest::gtest GTest::gtest_main stdc++fs dl ${PCIACCESS_LIBRARIES})
+else()
+  target_link_libraries(runCoreUnitTests PRIVATE ${ROCPROFILER_TARGET} ${AQLPROFILE_LIB}
+    hsa-runtime64::hsa-runtime64 c stdc++
+    GTest::gtest GTest::gtest_main stdc++fs dl ${PCIACCESS_LIBRARIES})
+endif()
 
 add_dependencies(tests runCoreUnitTests)
 install(TARGETS runCoreUnitTests RUNTIME DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/${PROJECT_NAME}/tests COMPONENT tests)
diff --git a/tests/unittests/profiler/CMakeLists.txt b/tests/unittests/profiler/CMakeLists.txt
index 7e08ace..17e86e2 100644
--- a/tests/unittests/profiler/CMakeLists.txt
+++ b/tests/unittests/profiler/CMakeLists.txt
@@ -83,11 +83,17 @@ target_include_directories(runProfilerUnitTests PRIVATE ${PROJECT_SOURCE_DIR}
 target_compile_definitions(runProfilerUnitTests
   PUBLIC AMD_INTERNAL_BUILD
   PRIVATE PROF_API_IMPL HIP_PROF_HIP_API_STRING=1 __HIP_PLATFORM_AMD__=1)
-
-target_link_libraries(runProfilerUnitTests PRIVATE rocprofiler_tool ${AQLPROFILE_LIB}
-  hsa-runtime64::hsa-runtime64
-  GTest::gtest GTest::gtest_main stdc++fs
-  ${PCIACCESS_LIBRARIES})
+if(NOT AQLPROFILE_LIB )
+  target_link_libraries(runProfilerUnitTests PRIVATE rocprofiler_tool
+    hsa-runtime64::hsa-runtime64
+    GTest::gtest GTest::gtest_main stdc++fs
+    ${PCIACCESS_LIBRARIES})
+else()
+  target_link_libraries(runProfilerUnitTests PRIVATE rocprofiler_tool ${AQLPROFILE_LIB}
+    hsa-runtime64::hsa-runtime64
+    GTest::gtest GTest::gtest_main stdc++fs
+    ${PCIACCESS_LIBRARIES})
+endif()
 
 add_dependencies(tests runProfilerUnitTests)
 install(TARGETS runProfilerUnitTests RUNTIME DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/${PROJECT_NAME}/tests COMPONENT tests)
-- 
2.39.1

